vm/devices/net/mana_driver/src/gdma_driver.rs

```diff
@@ -843,56 +850,126 @@ impl<T: DeviceBacking> GdmaDriver<T> {
         Ok(())
     }
 
-    async fn process_eqs_or_wait(&mut self) -> anyhow::Result<()> {
-        loop {
+    async fn process_eqs_or_wait_with_retry(
+        &mut self,
+    ) -> (bool, u128, u32, u32, u32, anyhow::Result<()>) {
+        let mut last_wait_result: anyhow::Result<()> = Ok(());
+        let mut elapsed: u128 = 0;
+        let mut eq_arm_count = 0;
+        let mut interrupt_wait_count = 0;
+        let mut interrupt_count = 0;
+        return loop {
             if self.process_all_eqs() {
-                return Ok(());
+                break (
+                    true,
+                    elapsed,
+                    interrupt_count,
+                    interrupt_wait_count,
+                    eq_arm_count,
+                    last_wait_result,
+                );
             }
-
             if !self.eq_armed {
-                tracing::trace!("arming eq");
+                eq_arm_count += 1;
                 self.eq.arm();
                 self.eq_armed = true;
                 // Check if the event arrived while arming.
                 if self.process_all_eqs() {
                     // Remove any pending interrupt events.
                     let _ = self.interrupts[0].as_mut().unwrap().wait().now_or_never();
-                    return Ok(());
+                    break (
+                        true,
+                        elapsed,
+                        interrupt_count,
+                        interrupt_wait_count,
+                        eq_arm_count,
+                        last_wait_result,
+                    );
                 }
             }
-            tracing::trace!("waiting for eq interrupt");
+            interrupt_wait_count += 1;
             let before_wait = std::time::Instant::now();
-            let wait_result = Self::wait_for_hwc_interrupt(
+            last_wait_result = Self::wait_for_hwc_interrupt(
                 self.interrupts[0].as_mut().unwrap(),
                 Some(&mut self.hwc_failure),
-                self.hwc_timeout_in_ms,
+                500,
```

Comment:
Optional: Create a variable like HWC_INTERRUPT_WAIT_TIME_IN_MS to store this 500.